xclim/indices/_agro.py

```diff
@@ -132,7 +133,7 @@ def corn_heat_units(
         )
     ) / 2
 
-    chu.attrs["units"] = ""
+    chu = chu.assign_attrs(units="")
     return chu
```

Comment:
Héhé this PR makes a clear point that shadowing variables isn't cool ;)

However, what's the issue with setting an attribute ?
```suggestion
    return chu.assign_attrs(units="")
```

Reply:
The thing with returning an object that is being modified "in-transit" is that `mypy` can't effectively interpret its output type. I think it needs to be defined so that typing inference can happen, then it can be returned.

Since this is modifying `chu` here, maybe we could declare the type of that variable and see what happens? I'll test that out.